File: app/src/main/java/com/worldventures/dreamtrips/social/ui/profile/service/command/RemoveFriendFromCircleCommand.java
package com.worldventures.dreamtrips.social.ui.profile.service.command;


import com.worldventures.core.model.Circle;
import com.worldventures.core.model.User;
import com.worldventures.core.service.command.api_action.ApiActionCommand;
import com.worldventures.dreamtrips.R;
import com.worldventures.dreamtrips.api.circles.RemoveFriendsFromCircleHttpAction;

import java.util.Collections;

import io.techery.janet.command.annotations.CommandAction;

@CommandAction
public class RemoveFriendFromCircleCommand extends ApiActionCommand<RemoveFriendsFromCircleHttpAction, Void> {

   private Circle circle;
   private int userId;

   public RemoveFriendFromCircleCommand(Circle circle, User friend) {
      this.circle = circle;
      this.userId = friend.getId();
   }

   public int getUserId() {
      return userId;
   }

   public Circle getCircle() {
      return circle;
   }

   @Override
   protected RemoveFriendsFromCircleHttpAction getHttpAction() {
      return new RemoveFriendsFromCircleHttpAction(circle.getId(), Collections.singletonList(userId));
   }

   @Override
   protected Class<RemoveFriendsFromCircleHttpAction> getHttpActionClass() {
      return RemoveFriendsFromCircleHttpAction.class;
   }

   @Override
   public int getFallbackErrorMessage() {
      return R.string.error_failed_to_remove_user_from_circle;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/social/service/profile/command/RemoveFriendFromCircleCommand.kt
package com.worldventures.dreamtrips.social.service.profile.command

import com.worldventures.core.model.Circle
import com.worldventures.core.model.User
import com.worldventures.core.service.command.api_action.ApiActionCommand
import com.worldventures.dreamtrips.R
import com.worldventures.dreamtrips.api.circles.RemoveFriendsFromCircleHttpAction

import io.techery.janet.command.annotations.CommandAction

@CommandAction
class RemoveFriendFromCircleCommand(val circle: Circle, friend: User) : ApiActionCommand<RemoveFriendsFromCircleHttpAction, Void>() {

   val userId = friend.id

   override fun getHttpAction() = RemoveFriendsFromCircleHttpAction(circle.id, listOf(userId))

   override fun getHttpActionClass() = RemoveFriendsFromCircleHttpAction::class.java

   override fun getFallbackErrorMessage() = R.string.error_failed_to_remove_user_from_circle
}
